core/src/main/java/nl/nn/adapterframework/core/PipeLineExit.java

```diff
@@ -72,30 +73,35 @@ public String getName() {
 		return getPath();
 	}
 
-
-	public String getState() {
-		return state;
-	}
-
 	@IbisDoc({"the exit state defines possible exists to the pipeline. the state can be one of the following: <code>success, error</code>", ""})
 	public void setState(String newState) {
 		state = newState;
 	}
-
-	public int getExitCode() {
-		return exitCode;
+	public String getState() {
+		return state;
 	}
 
 	@IbisDoc({"http statuscode e.g. <code>500</code>", "200"})
 	public void setCode(String code) {
 		this.exitCode = Integer.parseInt(code);
 	}
-	public boolean getEmptyResult() {
-		return emptyResult;
+	public int getExitCode() {
+		return exitCode;
 	}
 
 	@IbisDoc({"when using restlistener and set to <code>true</code>, this removes the output and shows a blank page, the output is still logged in the ladybug testtool", "false"})
 	public void setEmpty(String b) {
 		emptyResult = Boolean.parseBoolean(b);
 	}
+	public boolean getEmptyResult() {
+		return emptyResult;
+	}
+	// TODO: validate the output by looking at the elementName
+	@IbisDoc({"configures exit specific element reference in open api schema. If it is not set, but the responseRoot attribute is set in the validator, for the exit code 200 it will use the first element in the responseRoot and for the other codes last element of the responseRoot will be used.", ""})
```

Comment:
The choice of the 'first' or 'last' responseRoot value configured in the validator should be based on exit.state, instead of exit.code.

Please change the IbisDoc to:
Configures the responseRoot in the OpenAPI schema for this exit. If not set, the responseRoot value of the validator will be used. If that contains multiple (comma separated) values, the first will be used for the exits with state 'success', the last for the other exits.